### PR TITLE
Add numerical comparison and equality expression support

### LINKED ISSUE #22409: Add numerical comparison and equality expressions
**Description:**
Add:
- Numerical comparison expression: https://ballerina.io/spec/lang/2020R1/#numerical-comparison-expr
- Equality expression: https://ballerina.io/spec/lang/2020R1/#equality-expr

Both are binary expressions. So the implementation would be similar for both.

Have a look at the existing binary expressions (additive-expr. e.g: `a + b`) for sample impl.

### PR DESCRIPTION
**Pull request by lochana-chathura**
Fixes #22409 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/tree/SyntaxKind.java [hunk 1]
```diff
@@ -77,12 +77,16 @@
...
+    LT_EQUAL_TOKEN(559, "<="),
     GT_TOKEN(560, ">"),
     EQUAL_GT_TOKEN(561, "=>"),
     QUESTION_MARK_TOKEN(562, "?"),
     PIPE_TOKEN(563, "|"),
-
+    GT_EQUAL_TOKEN(564, ">="),
+    EXCLAMATION_MARK_TOKEN(565, "!"),
+    EM_EQUAL_TOKEN(566, "!="),
+    EM_DOUBLE_EQUAL_TOKEN(567, "!=="),
```
Comment (SupunS) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/tree/SyntaxKind.java:88]: Shall we rename this to `NOT_EQUAL` and `NOT_DOUBLE_EQUAL`?

File: compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/ParserTestUtils.java [hunk 2]
```diff
@@ -348,13 +348,23 @@ private static SyntaxKind getNodeKind(String kind) {
             case "PERCENT_TOKEN":
                 return SyntaxKind.PERCENT_TOKEN;
             case "EQUAL_LT_TOKEN":
-                return SyntaxKind.EQUAL_LT_TOKEN;
+                return SyntaxKind.LT_EQUAL_TOKEN;
```
Comment (SupunS) [inline compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/ParserTestUtils.java:351]: case should be `case "LT_EQUAL_TOKEN"`

Comment (lochana-chathura) [reply compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/ParserTestUtils.java]: I have added it below. so this case was removed.

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/tree/SyntaxKind.java (hunk 1 above)
Comment (lochana-chathura) [reply compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/tree/SyntaxKind.java:88]: sure.

**Comment by lochana-chathura** [comment]
There were few test cases already for operator precedency. Added new one covering all levels.

**Pull request merged**